frame/support/src/weights/weight_v2.rs

```diff
@@ -0,0 +1,480 @@
+		return self.0.weigh_data(args)
+	}
+}
+
+impl<T> ClassifyDispatch<T> for (Weight, DispatchClass) {
+	fn classify_dispatch(&self, _: T) -> DispatchClass {
+		self.1
+	}
+}
+
+impl<T> PaysFee<T> for (Weight, DispatchClass) {
+	fn pays_fee(&self, _: T) -> Pays {
+		Pays::Yes
+	}
+}
+
+impl<T> WeighData<T> for (Weight, Pays) {
+	fn weigh_data(&self, args: T) -> Weight {
+		return self.0.weigh_data(args)
+	}
+}
+
+impl<T> ClassifyDispatch<T> for (Weight, Pays) {
+	fn classify_dispatch(&self, _: T) -> DispatchClass {
+		DispatchClass::Normal
+	}
+}
+
+impl<T> PaysFee<T> for (Weight, Pays) {
+	fn pays_fee(&self, _: T) -> Pays {
+		self.1
+	}
+}
+
+impl From<(Option<Weight>, Pays)> for PostDispatchInfo {
+	fn from(post_weight_info: (Option<Weight>, Pays)) -> Self {
+		let (actual_weight, pays_fee) = post_weight_info;
+		Self { actual_weight, pays_fee }
+	}
+}
+
+impl From<Option<Weight>> for PostDispatchInfo {
+	fn from(actual_weight: Option<Weight>) -> Self {
+		Self { actual_weight, pays_fee: Default::default() }
+	}
+}
+
+impl<T> WeighData<T> for Weight {
+	fn weigh_data(&self, _: T) -> Weight {
+		return *self
+	}
+}
+
+impl<T> ClassifyDispatch<T> for Weight {
+	fn classify_dispatch(&self, _: T) -> DispatchClass {
+		DispatchClass::Normal
+	}
+}
+
+impl<T> PaysFee<T> for Weight {
+	fn pays_fee(&self, _: T) -> Pays {
+		Pays::Yes
+	}
+}
+
+impl<T> ClassifyDispatch<T> for (Weight, DispatchClass, Pays) {
+	fn classify_dispatch(&self, _: T) -> DispatchClass {
+		self.1
+	}
+}
+
+impl core::fmt::Display for Weight {
+	fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+		write!(f, "Weight(ref_time: {})", self.ref_time)
+	}
+}
+
+impl Bounded for Weight {
+	fn min_value() -> Self {
+		Zero::zero()
+	}
+	fn max_value() -> Self {
+		Self::MAX
+	}
+}
+
+impl AddAssign for Weight {
+	fn add_assign(&mut self, other: Self) {
+		*self = Self { ref_time: self.ref_time + other.ref_time };
+	}
+}
+
+impl SubAssign for Weight {
+	fn sub_assign(&mut self, other: Self) {
+		*self = Self { ref_time: self.ref_time - other.ref_time };
+	}
+}
+
+impl sp_runtime::traits::Printable for Weight {
+	fn print(&self) {
+		self.ref_time().print()
+	}
+}
+
+// Re-export common functions so you do not need to import trait.
+impl Weight {
+	pub const fn zero() -> Self {
+		Self { ref_time: 0 }
+	}
+
+	pub const fn one() -> Self {
+		Self { ref_time: 1 }
+	}
+
+	pub fn saturating_add(self, rhs: Self) -> Self {
+		<Self as Saturating>::saturating_add(self, rhs)
+	}
+
+	pub fn saturating_sub(self, rhs: Self) -> Self {
+		<Self as Saturating>::saturating_sub(self, rhs)
+	}
+
+	pub fn saturating_mul(self, rhs: Self) -> Self {
+		<Self as Saturating>::saturating_mul(self, rhs)
+	}
+
+	pub fn checked_add(&self, rhs: &Self) -> Option<Self> {
+		<Self as CheckedAdd>::checked_add(self, rhs)
+	}
+
+	pub fn checked_sub(&self, rhs: &Self) -> Option<Self> {
+		<Self as CheckedSub>::checked_sub(self, rhs)
+	}
+}
+
+// TODO: Eventually remove these
```

Comment:
In this PR or in the future?

Reply:
in the future with weight v2 stuff.

basically, i don't want to let people cheat by passing a single u64 into the weight, and this allows that at the moment